### PR TITLE
feat: implement version/index.html page

### LINKED ISSUE #117: Implement "Older versions" link in multi-version drop down
💡 **Idea**
The main idea behind this issue is to have a link named "Older versions" in the drop down button for the multi-version documentation. This link must only appear after more than 3 stable versions are found for a project.

The idea is simple, but the implementation is a bit tricky. The main issue is that rendered documentation are not aware about the version context. This implies, that, we need to create the page and inject it after the content has been rendered.

1. The `ansys-sphinx-theme` needs to generate a page that is not linked to any other one. This page must be named `releases.html` and should contain a `Jinja` section to be rendered after the documentation has been generated. This is the only way we can ensure that the documentation sections and style match the latest stable release of a project.
2. In the `doc-deploy-stable`, it renders previous Jinja context with the name and URL for each existing version. 
3. The HTML page is placed in the `gh-pages` in the `release/index.html` path.

### PR DESCRIPTION
**Pull request by jorgepiloto**
Fix #117 by adding a `version/index.html` page that reads the content of the `versions.json` file and creates a table showing all the versions available in the documentation.

### CODE REVIEW

File: doc-deploy-dev/versions.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        document.documentElement.innerHTML = html;
+      }),
+    fetch("versions.json")
+      .then((res) => res.json())
+      .then((data) => {
+        var table = "<h2>Older versions</h2>";
+        table += "<table class='table'>";
+        table += "<th> versions </th>";
+        table += "<th> url </th>";
+        for (let versions of data) {
```
Comment (jorgepiloto) [inline doc-deploy-dev/versions.js:20]: Let's play also with a variable named `ith_version` so we can add the `row-odd` and `row-even` classes to the right rows in the table. This guarantees that the final style of the table matches the one from the ansys-sphinx-theme.

**Comment by Revathyvenugopal162** [comment]
keeping the ss of output with script for reference.
### expected output

image

**Comment by jorgepiloto** [comment]
Thanks for adding that script, @Revathyvenugopal162. Let me integrate it inside of the actions.

**Comment by jorgepiloto** [comment]
I just realized that we are forced to collect the directories next to the 'version/index.html' to check which versions are available. The 'versions.json' file should only contain the latest three and a link to the 'version/index.html' page with the title `Older versions`.

**Comment by Revathyvenugopal162** [comment]
> I just realized that we are forced to collect the directories next to the 'version/index.html' to check which versions are available. The 'versions.json' file should only contain the latest three and a link to the 'version/index.html' page with the title `Older versions`.

Yes true, i was just reading the json to write the table.
But reading directories with js will give duplicate the stable version and its actual version number.because the stable doc have the duplicated files in the latest version folder i think(0.64 and stable have same thing in mapdl)

File: doc-deploy-dev/action.yml [hunk 2]
```diff
@@ -211,6 +211,144 @@ runs:
...
+    - name: "Update all references to point to 'stable/_static' or 'dev/_static'"
+      shell: bash
+      run: |
+        if [[ -f 'version/stable/index.html' ]]; then
+          sed -i "s|_static|stable/_static|g" version/index.html
+        else
+          sed -i "s|_static|dev/_static|g" version/index.html
+        fi
+
+    - name: "Declare the script to read the 'versions.json' file"
```
Comment (jorgepiloto) [inline doc-deploy-dev/action.yml:240]: No need to scrap anything. The repository gets checkout first of executing any logic. Therefore, just a list of desired folders versions and pass it to the desired JS code.

**Review by jorgepiloto — COMMENTED** [review]
Leaving some comments before I forget about cleaning this implementation.

File: doc-deploy-dev/action.yml (hunk 2 above)
Comment (jorgepiloto) [reply doc-deploy-dev/action.yml:245]: Thank you, Jorge from the past.

**Review by RobPasMue — APPROVED** [review]
Looks great @jorgepiloto - let's just modify the description of the action and add my minor typo suggestion =)

File: _doc-gen-versions-page/action.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+description: |
+  This action generates the ``version/index.html`` page. This page collects all
+  the releases available for a library. This feature allows to not overload the
+  content of the dropdown button for the multi-version.
+
+  .. warning::
+
+      This is a private action and its meant to be reused by ``doc-deploy-*``
+      actions.
+
```
[the remark this answers is not in the document]
Comment (RobPasMue) [reply _doc-gen-versions-page/action.yml:13]: I mean it is not the "Doc deploy dev action"

**Pull request merged**